datafusion/core/src/physical_plan/file_format/mod.rs

```diff
@@ -183,17 +183,27 @@ struct FileGroupsDisplay<'a>(&'a [Vec<PartitionedFile>]);
 
 impl<'a> Display for FileGroupsDisplay<'a> {
```

Comment:
This implementation also avoids allocating strings and Vecs, which I don't think is a particularly important feature, but it is a nice benefit.